### ISSUE TITLE
[kws] tc-resnet reproduce

### ISSUE THREAD

**Issue by lippman1125** [google-research/google-research#371]
@rybakov 
In the paper [streaming keyword spotting on mobile devices](https://arxiv.org/abs/2005.06720).
```
3.8 Temporal Convolution ResNet(TC-ResNet)
`"To improve accuracy of this model we increase number of parameters from 305k to 365k".`
```
The TC-ResNet14-1.5 is 305K, so I want to know the config of 365k model.Would you supply me with the config of 365k model.

I tried the default hyper parameters of author as follow:
```
$CMD_TRAIN \
--data_url '' \
--data_dir $DATA_PATH/ \
--train_dir $MODELS_PATH/tc_resnet/ \
--mel_upper_edge_hertz 7000 \
--how_many_training_steps 10000,10000,10000 \
--learning_rate 0.1,0.01,0.001 \
--window_size_ms 30.0 \
--window_stride_ms 10.0 \
--mel_num_bins 40 \
--dct_num_features 40 \
--resample 0.15 \
--alsologtostderr \
--train 1 \
--optimizer 'momentum' \
--lr_schedule 'linear' \
--use_spec_augment 1 \
--time_masks_number 2 \
--time_mask_max_size 10 \
--frequency_masks_number 2 \
--frequency_mask_max_size 5 \
tc_resnet \
--kernel_size '(9,1)' \
--channels '24, 36, 36, 48, 48, 72, 72' \
--debug_2d 0 \
--pool_size '' \
--pool_stride 0 \
--l2_weight_decay 0.001 \
--bn_momentum 0.997 \
--bn_center 1 \
--bn_scale 1 \
--bn_renorm 0 \
--dropout 0.5
```

But I can not reproduce the accuracy(96.6% in paper) on V1 dataset.

Another question, why don't you add weight_decay in tc_resnet?
https://github.com/google-research/google-research/blob/3a228f1d94e6369072b2aeb3ae81644a95c53d93/kws_streaming/models/tc_resnet.py#L139-L145

**Comment by rybakov** [comment]
lippman1125@ I reverted a code refactor (which was compatible with eager mode only) please update your code with the latest version. I trained tc_resent model on data set v1 and get 97.1%

I use 365K model:
$CMD_TRAIN \
--data_url '' \
--data_dir $DATA_PATH/ \
--train_dir $MODELS_PATH/tc_resnet/ \
--mel_upper_edge_hertz 7000 \
--how_many_training_steps 20000,20000,20000,20000 \
--learning_rate 0.1,0.05,0.02,0.01 \
--window_size_ms 40.0 \
--window_stride_ms 20.0 \
--mel_num_bins 40 \
--dct_num_features 40 \
--resample 0.15 \
--alsologtostderr \
--train 1 \
--optimizer 'momentum' \
--lr_schedule 'exp' \
--use_spec_augment 1 \
--time_masks_number 2 \
--time_mask_max_size 10 \
--frequency_masks_number 2 \
--frequency_mask_max_size 5 \
tc_resnet \
--channels '50, 50, 50, 50, 50, 72, 72' \
--debug_2d 0 \
--pool_size '' \
--pool_stride 0 \
--bn_momentum 0.997 \
--bn_center 1 \
--bn_scale 1 \
--bn_renorm 0 \
--dropout 0.2  

About reproducing accuracy 96.6% of model TC-ResNet14-1.5, 305K on data set v1. I reimplemented tc resnet in keras in kws_streaming lib: the model topology is correct, but there can be some differences in specch feature extractor and model weights initialization, because the original TC-ResNet14-1.5 was implemented with slim library (my implementation is based on it but it is not absolutely the same). So you will need to run some hyper parameters optimization.
With below comand I get 96.69% (as I mentioned earlier, please use the latest version of kws lib):

$CMD_TRAIN \
--data_url '' \
--data_dir $DATA_PATH/ \
--train_dir $MODELS_PATH/tc_resnet_305/ \
--mel_upper_edge_hertz 7000 \
--how_many_training_steps 20000,20000,20000,20000 \
--learning_rate 0.1,0.05,0.02,0.01 \
--window_size_ms 30.0 \
--window_stride_ms 10.0 \
--mel_num_bins 40 \
--dct_num_features 40 \
--resample 0.15 \
--alsologtostderr \
--train 1 \
--optimizer 'momentum' \
--lr_schedule 'linear' \
--use_spec_augment 1 \
--time_masks_number 2 \
--time_mask_max_size 10 \
--frequency_masks_number 2 \
--frequency_mask_max_size 5 \
tc_resnet \
--kernel_size '(9,1)' \
--channels '24, 36, 36, 48, 48, 72, 72' \
--debug_2d 0 \
--pool_size '' \
--pool_stride 0 \
--bn_momentum 0.997 \
--bn_center 1 \
--bn_scale 1 \
--bn_renorm 0 \
--dropout 0.5

**Comment by rybakov** [comment]
About weight decay - I did not see much accuracy improvements with it, but on the other hand I did not run exhaustive hyper parameter optimization, so please feel free to explore it.
Please let me know if you have more questions, thank you for evaluating this lib!

**Comment by lippman1125** [comment]
@rybakov 
Thanks for your reply, I have reproduced the results as follow:

| gscd v1 dataset| gscd v1 dataset | gscd v2 dataset|gscd v2 dataset|
|--|--|--|--|
|tc_resnet_305k | tc_resnet_365k | tc_resnet_305k | tc_resnet_365k|
|96.82% | 97.21% | 97.18% | 97.63%|

Each model I trained 15 times and picked the best。

**Comment by rybakov** [comment]
lippman1125@ thank you for reporting your results - it is great that all your numbers are higher than what I reported in [paper](https://arxiv.org/pdf/2005.06720.pdf). In my case, I trained every model 4 times and reported averaged accuracy, that is why my results are more reserved.
